### PR TITLE
fix: Fix for materializing entityless feature views in Snowflake

### PR DESCRIPTION
**Pull request by JohnLemmonMedely**
**What this PR does / why we need it**: When using an entityless featureview with a Snowflake offline store it fails to materialize correctly.

**Which issue(s) this PR fixes**:

Fixes # 3960

### CODE REVIEW

File: sdk/python/feast/infra/materialization/snowflake_engine.py [hunk 1]
```diff
@@ -274,7 +274,11 @@ def _materialize_one(
 
             fv_latest_values_sql = offline_job.to_sql()
 
-            if feature_view.entity_columns:
+            if (
+                feature_view.entity_columns[0].name == DUMMY_ENTITY_ID
+            ):  # entityless Feature View's placeholder entity
+                entities_to_write = 1
+            else:
```
Comment (JohnLemmonMedely) [inline sdk/python/feast/infra/materialization/snowflake_engine.py:281]: Entityless feature views do have an entity column, it's just that it's a dummy value.

**Comment by franciscojavierarceo** [comment]
@JohnLemmonMedely can you provide more context on this one? Do we support entityless feature views outside of this? Or is this a bit of a hack?

**Comment by JohnLemmonMedely** [comment]
> @JohnLemmonMedely can you provide more context on this one? Do we support entityless feature views outside of this? Or is this a bit of a hack?

Yes, I hit this bug when setting up a feature view [like this](https://docs.feast.dev/getting-started/concepts/feature-view#feature-views-without-entities). It looks like the documentation calls this "Feature views without entities" but the code refers to it as "entityless feature views".

In this case the original code was checking for an entityless feature view by checking if `entity_columns` is truthy. I'm not sure if this was once a valid check or not but today in Feast entityless feature views are given a dummy entity_column instead.

**Comment by JohnLemmonMedely** [comment]
@franciscojavierarceo Does my above comment answer your question? Anything else I need to do before merging this in?

**Review by franciscojavierarceo — APPROVED** [review]
this lgtm. your pr title lint failed

**Comment by JohnLemmonMedely** [comment]
> this lgtm. your pr title lint failed

Thanks @franciscojavierarceo, I think I fixed the PR title earlier so the check is passing now.

**Pull request merged**